[FIX] web: adapt bottom sheet inner layouts

Brieuc-brd opened the pull request:
This commit fixes several layout issues related to the content of the bottom sheet.

task-5087158




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/227230.png)](https://mergebot.odoo.com/odoo/odoo/pull/227230)

stefanorigano commented on addons/web/static/src/core/bottom_sheet/bottom_sheet.scss:
```diff
@@ -274,12 +319,23 @@
         .dropdown-item {
-            &.active, &.selected {
+            &.selected {
+                @extend %BottomSheet-Entry-active;
+                @extend %BottomSheet-Entry-selected;
```
Both states? Sounds wrong... it may be the cause of this issue (calendar view)
<img width="465" height="492" alt="image" src="https://github.com/user-attachments/assets/2c4093d2-e581-4ae6-bcb3-66f3ad44dac8" />

Brieuc-brd replied:
Good catch 👍 
The issue comes to [my selector](https://github.com/odoo/odoo/pull/227230/files#diff-d0207dbfa92b174873366c34ae08ac9a10f6c387c5534d7ce33f4c1bb7170237R335-R336) that was not strong enough to target the "Progress" items, and so the style is also applied when it should not, like here in calendar.
Unfortunately, we should need a flag class to trigger this design correctly. 🥲
As we can't do this in stable, I'll revert this change.

rfr-odoo replied:
Also there is the `project_task_state_selection_menu` dropdown that is impacted
<img width="360" height="626" alt="image" src="https://github.com/user-attachments/assets/9d8422d6-eac3-48f5-b92d-7d877b05eaad" />

stefanorigano commented on addons/web/static/src/core/colorlist/colorlist.scss:
```diff
@@ -1,5 +1,58 @@
+            overflow: hidden;
+            aspect-ratio: var(--ColorListField-Entry-aspectRatio, 1);
+            border-radius: var(--ColorListField-Entry-borderRadius, 100%);
+
```
ops, empty line

stefanorigano commented on addons/web/static/src/core/colorlist/colorlist.scss:
```diff
@@ -1,5 +1,58 @@
+            border-radius: var(--ColorListField-Entry-borderRadius, #{$border-radius});
+            overflow: hidden;
+            aspect-ratio: var(--ColorListField-Entry-aspectRatio, 1);
+            border-radius: var(--ColorListField-Entry-borderRadius, 100%);
```
border-radius is defined twice (see L14)

stefanorigano commented on addons/web/static/src/core/colorlist/colorlist.scss:
```diff
@@ -1,5 +1,58 @@
+
+            &::after {
+                @include o-position-absolute(0, 0 , 0, 0);
+                color: white;
```
Be aware of the not-colored badge
<img width="434" height="192" alt="image" src="https://github.com/user-attachments/assets/8cc3c031-fb73-497a-bcb9-73d106bba476" />
... and followup for DM, now close to the merge https://github.com/odoo/odoo/pull/227123

Brieuc-brd commented:
Thanks for your feedbacks @stefanorigano  and @rfr-odoo 👍 
I've updated the branch, here's the diff : 
https://github.com/odoo/odoo/compare/8c024802b9f147267630acd78c24bf023ff2ae02..1cb0846f48933e0c3bddda913088f54d759f6a55

> Can you check there is a problem with many2many_tags widget

It's now fixed ✅ 
Also, shouldn’t it be a bottom sheet element as well? Not in this PR, but I guess it’s something we should address on master 🙂 

What do you think ?

rfr-odoo commented:
> Also, shouldn’t it be a bottom sheet element as well? Not in this PR, but I guess it’s something we should address on master 🙂

It's because it's not a `DropDown` but a simple `Popover` and yes in master we will continue the work to have more BottomSheet.

rfr-odoo commented:
There is a small problem into the `o_color_picker_cell`:
In iPhoneSE the two colors on the left are hidden
<img width="1603" height="793" alt="image" src="https://github.com/user-attachments/assets/4a840ab9-ac35-46be-a58c-52d51cbfa47b" />
(screen from Project > Configuration > Tags)

Brieuc-brd commented:
The PR is ready for tech. review, @rfr-odoo  could you have a look ?
Thanks in advance! 🙏

rfr-odoo reviewed (approved):
@robodoo r+
